### PR TITLE
(EAI-346): Radiant integration

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-346

## Changes

- Clients to connect to embedding and LLM models hosted on radiant

## Notes

- Support MongoDB corp security cookie if necessary
- Before merge, wait for https://jira.mongodb.org/plugins/servlet/desk/portal/48/KANOPY-708

### CODE REVIEW

File: packages/chatbot-server-mongodb-public/.env.example [hunk 1]
```diff
@@ -11,3 +11,6 @@ MONGODB_DATABASE_NAME=<docs-chatbot-ENV>
 VECTOR_SEARCH_INDEX_NAME=<index_name>
 NODE_ENV=development
 ALLOWED_ORIGINS=http://example.com,http://localhost:5173
+RADIANT_BASE_URL="<some radiant uri>"
+RADIANT_API_KEY="<some radiant API key>"
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/.env.example:15]: I get the `RADIANT` naming for differentiating in the tests but longer term the idea would be that you specify the radiant URL/API key for `OPENAI_ENDPOINT` and `OPENAI_API_KEY` right?

**Review by nlarew — APPROVED** [review]

File: packages/chatbot-server-mongodb-public/.env.example (hunk 1 above)
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/.env.example:15]: > I get the RADIANT naming for differentiating in the tests but longer term the idea would be that you specify the radiant URL/API key for OPENAI_ENDPOINT and OPENAI_API_KEY right?

i'm not sure if it'd matter necessarily. why use the same var?

File: packages/mongodb-chatbot-server/src/routes/generateResponse.ts [hunk 2]
```diff
@@ -296,6 +296,7 @@ export async function streamGenerateResponse({
         if (toolReferences) {
           outputReferences.push(...toolReferences);
         }
+        console.log([...llmConversation, ...newMessages]);
```
Comment (mongodben) [inline packages/mongodb-chatbot-server/src/routes/generateResponse.ts:299]:
```suggestion
```

**Comment by mongodben** [comment]
closing this PR b/c we've decided not to proceed with the radiant integration for the time being. 

the code in this PR actually does work in kanopy when you update the:
1. `OPENAI_API_KEY` and `OPENAI_ENDPOINT` in kanopy
2. set up a sidecar service to work with istio
3. also not in the PR, but you'd need to set up the vanity domain https://knowledge.staging.corp.mongodb.com/ to work with the new istio-backed app. this'd be a kanopy ticket

for more information on setting up radiant across namespaces, refer to https://docs.google.com/document/d/1U_t1D4999b59FsUmhYMJyahCVttx1D_bs-v6jlYQqsw/edit#heading=h.trrflxo7p9nk (the content of which might move to the wiki at some point)

**Comment by mongodben** [comment]
closing this PR b/c we've decided not to proceed with the radiant integration for the time being. 

the code in this PR actually does work in kanopy when you update the:
1. `OPENAI_API_KEY` and `OPENAI_ENDPOINT` in kanopy
2. set up a sidecar service to work with istio
3. also not in the PR, but you'd need to set up the vanity domain https://knowledge.staging.corp.mongodb.com/ to work with the new istio-backed app. this'd be a kanopy ticket

for more information on setting up radiant across namespaces, refer to https://docs.google.com/document/d/1U_t1D4999b59FsUmhYMJyahCVttx1D_bs-v6jlYQqsw/edit#heading=h.trrflxo7p9nk (the content of which might move to the wiki at some point)

**Pull request closed without merging**